documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,472 @@
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "timeout"
+        in: "query"
+        description: "Maximum duration of the GC Dump"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "timeout"
+        in: "query"
+        description: "Maximum duration of the GC Dump"
+        required: false
+        type: "integer"
+        default: "30"
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /cpuprofile:
+    get:
+      tags:
+      - "traces"
+      operationId: "getdefaultcputrace"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "duration"
+        in: "query"
+        description: "Duration of the CPU trace"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /cpuprofile/{uid}:
+    get:
+      tags:
+      - "traces"
+      operationId: "getcputrace"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "duration"
+        in: "query"
+        description: "Duration of the CPU trace"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /trace:
```

Comment:
we should consider a GET based trace call with the same defaults as the default /logs call.